app/models/manageiq/providers/openstack/cloud_manager.rb

```diff
@@ -56,7 +56,25 @@ class ManageIQ::Providers::Openstack::CloudManager < ManageIQ::Providers::CloudM
   after_save :refresh_parent_infra_manager
 
   def self.params_for_create
-    @params_for_create ||= {
+    static_params_for_create.deep_clone.tap do |params|
+      params[:fields].detect { |f| f[:id] == 'provider_id' }[:options] = provider_id_options
+    end
+  end
+
+  private_class_method def self.provider_id_options
+    (t = ManageIQ::Providers::Openstack::Provider)
+      .order(t.arel_table[:name].lower)
```

Comment:
@agrare As discussed, here is the extra fix to order by name.  I think there are also needs to be an RBAC tenancy check here.  Thoughts on how best to do this?  Is it just `RBAC.filtered(...)`?

Reply:
Yeah `Rbac.filtered` should do it